cns/restserver/internalapi_test.go

```diff
@@ -0,0 +1,189 @@
+	for i := 0; i < secondaryIpCount; i++ {
+		ipaddress := "10.0.0." + strconv.Itoa(startingIndex)
+		secIpConfig := newSecondaryIPConfig(ipaddress, 32)
+		ipId, err := uuid.NewUUID()
+
+		if (err != nil) {
+			t.Errorf("Failed to generate UUID for secondaryipconfig, err:%s", err)
+			t.Fatal(err)
+		}
+		secondaryIPConfigs[ipId.String()] = secIpConfig
+		startingIndex++
+	}
+
+	createAndValidateNCRequest(t, secondaryIPConfigs)
+
+	// now Scale down, delete 3 ipaddresses from secondaryIpConfig req
+	fmt.Println("Validate Scalesown")
+	var count = 0
+	for ipid, _ := range secondaryIPConfigs {
+		delete(secondaryIPConfigs, ipid)
+		count++
+
+		if count > secondaryIpCount {
+			break
+		}
+	}
+
+	createAndValidateNCRequest(t, secondaryIPConfigs)
+
+	// Cleanup all SecondaryIps
+	fmt.Println("Validate no SecondaryIpconfigs")
+	for ipid, _ := range secondaryIPConfigs {
+		delete(secondaryIPConfigs, ipid)
+	}
+
+	createAndValidateNCRequest(t, secondaryIPConfigs)
+
+	return nil
+}
+
+func createAndValidateNCRequest(t *testing.T, secondaryIPConfigs map[string]cns.SecondaryIPConfig) {
+	req := generateNetworkContainerRequest(secondaryIPConfigs)
+	returnCode := svc.CreateOrUpdateNetworkContainerInternal(req)
+	if returnCode != 0 {
+		t.Errorf("Failed to createNetworkContainerRequest, req: %+v, err: %d", req, returnCode)
+		t.Fatal()
+	}
+	validateNetworkRequest(t, req)
+}
+
+// Validate the networkRequest is persisted.
+func validateNetworkRequest(t *testing.T, req cns.CreateNetworkContainerRequest) {
+	containerStatus := svc.state.ContainerStatus[req.NetworkContainerid]
+
+	if containerStatus.ID != req.NetworkContainerid {
+		t.Errorf("Failed as NCId is not persisted, expected:%s, actual %s", req.NetworkContainerid, containerStatus.ID)
+		t.Fatal()
+	}
+
+	actualReq := containerStatus.CreateNetworkContainerRequest
+	if actualReq.NetworkContainerType != req.NetworkContainerType {
+		t.Errorf("Failed as ContainerTyper doesnt match, expected:%s, actual %s", req.NetworkContainerType, actualReq.NetworkContainerType)
+		t.Fatal()
+	}
+
+    if actualReq.IPConfiguration.IPSubnet.IPAddress != req.IPConfiguration.IPSubnet.IPAddress {
+		t.Errorf("Failed as Primary IPAddress doesnt match, expected:%s, actual %s", req.IPConfiguration.IPSubnet.IPAddress, actualReq.IPConfiguration.IPSubnet.IPAddress)
+		t.Fatal()
+	}
+
+	// Validate Secondary ips are added in the PodMap
+	if len(svc.PodIPConfigState) != len(req.SecondaryIPConfigs) {
+		t.Errorf("Failed as Secondary IP count doesnt match in PodIpConfig state, expected:%d, actual %d", len(req.SecondaryIPConfigs), len(svc.PodIPConfigState))
+		t.Fatal()
+	}
+
+	var alreadyValidated = make(map[string]string)
+	for ipid, ipStatus := range svc.PodIPConfigState {
+		if ipaddress, found := alreadyValidated[ipid]; !found {
+			if secondaryIpConfig, ok := req.SecondaryIPConfigs[ipid]; !ok {
+				t.Errorf("PodIpConfigState has stale ipId: %s, config: %+v", ipid, ipStatus)
+				t.Fatal()
+			} else {
```

Comment:
else is not needed. the block above is an early exit from this func

Reply:
it is needed in case entry doesnt exist which is an expected scenario